src/main/java/core/basesyntax/TreeSetCharacters.java

```diff
@@ -17,7 +23,24 @@
  * Результат 2: acf</p>
  */
 public class TreeSetCharacters {
-    public String getUniqueCharacters(String fileName) {
-        return null;
+    public String getUniqueCharacters(String fileName) throws IOException {
+        Set<Character> set = new TreeSet<>();
+        try (FileReader fr = new FileReader(fileName)) {
+            Scanner scan = new Scanner(fr);
+            while (scan.hasNextLine()) {
+                String string = scan.nextLine();
+                for (char c : string.toCharArray()) {
+                    if (Character.isAlphabetic(c)) {
+                        set.add(Character.toLowerCase(c));
+                    }
+                }
+            }
+            StringBuilder result = new StringBuilder();
+            for (char c : set) {
+                result.append(c);
+            }
```

Comment:
can we just stop appending when we have reached the required size?

Reply:
Updated. Done